src/BaseExtension.php

```diff
@@ -450,7 +450,7 @@ public function addSnippet($location, $callback, $extraparameters = [])
             $callback = (string) $callback;
         }
 
-        $this->app['asset.queue.snippet']->add($location, $callback, $this->getName(), (array) $extraparameters);
+        $this->app['asset.queue.snippet']->add($location, [$this, $callback], $this->getName(), (array) $extraparameters);
     }
```

Comment:
Ah, you might want to check for a string first, as you can (and I already do) pass in a callback that is not in `Extension.php`.

Also probably worth checking if the string is a method in `$this` as is can also be parsed a pre-built html element.